gke-policies/policy/node_pool_use_cos.rego

```diff
@@ -0,0 +1,31 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# METADATA
+# title: Use Container-Optimized OS
+# description: GKE node pools should use Container-Optimized OS which is maintained by Google and optimized for running Docker containers with security and efficiency.
+# custom:
+#   group: Security
+package gke.policy.node_pool_use_cos
+
+default valid = false
+
+valid {
+  count(violation) == 0
+}
+
+violation[msg] {  
+  not startswith(lower(input.node_pools[pool].config.image_type), "cos")
```

Comment:
This is something really minor, but I'm wondering if we can be more specific and tolerate only `cos` and `cos_containerd`.

In a such case, you could use `in` keyword, example:

```
import future.keywords.in
violation[msg] {  
  not lower(input.node_pools[pool].config.image_type) in  {"cos", "cos_containerd"}
  ...
}
```

Reply:
Good point! I'll change it.
 
I was wondering which approach to go and ended up going for prefix in case some other "cos" flavour be created.
However, I agree it's more solid to be specific, and we could always add new "cos" versions in the future.